### ISSUE TITLE
Unable to process beacon beacons' - beacons.save does not work and fills /var/log/salt

### ISSUE THREAD

**Issue by golmaal** [saltstack/salt#58579]
**Description**
If beacons.save is called and there are no beacons to save, salt-minion ends up writing the following to its config file - 

cat /etc/salt/minion.d/beacons.conf
beacons:
  beacons: {}

After salt-minion is restarted, it continuously start to print the "Unable to process beacon" message.

**Setup**
CentOS 8.1 docker container
salt bootstrap based git v3001.1 install

**Steps to Reproduce the behavior**
Start centos:8.1.1911 in a container
Install python3 and git
Install salt minion and master as following - 
  sudo sh bootstrap-salt.sh -M git v3001.1
Accept minion's key and call beacons.save
salt-call beacons.save
Now after restart of salt-minion, it starts to print that annoying message. 

The fix would probably be to leave the beacons.conf file empty instead of saving the following in it - 

beacons:
  beacons: {}

**Versions Report**
```
Salt Version:
           Salt: 3001.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.11.2
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 1.0.0
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: Not Installed
   pycryptodome: 3.9.8
         pygit2: Not Installed
         Python: 3.6.8 (default, Apr 16 2020, 01:36:27)
   python-gnupg: Not Installed
         PyYAML: 5.3.1
          PyZMQ: 19.0.2
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.2

System Versions:
           dist: centos 8 Core
         locale: UTF-8
        machine: x86_64
        release: 4.18.0-193.19.1.el8_2.x86_64
         system: Linux
        version: CentOS Linux 8 Core
```

**Comment by dkfsalt** [comment]
Suggest we change title to "Unable to process beacon beacons' - beacons.save does not work and fills /var/log/salt"

## SUMMARY

This is a critical failure of the beacons module.  Those users that are doing a highstate every hour will have an outage of their beacons (after a minion restart) up until the next time they run highstate (although their /var/log/salt partition is going to fill up with these error messages).

@golmaal was just seeing the symptoms of the beacons.save() bug.

## REPRODUCE

The contents of _/etc/salt/minion.d/beacons.conf_ is the symptom rather than the root cause.  Your beacons.conf file gets into this state the moment that you use _beacons.save_....

I'm reproducing this on CentOS 7, salt-minion 3001.1

1. Start with a clean slate
```
systemctl stop salt-minion; rm -rf /etc/salt/minion.d/beacons.conf /var/log/salt/minion /var/cache/salt/minion/*; systemctl start salt-minion; sleep 10; grep -c 'Unable to process beacon beacons' /var/log/salt/minion
0  <-- no errors...
```

2. Configure some beacons
```
# salt-call state.apply beacon.setup
local:
----------
          ID: install mysql
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 20:13:48.676895
    Duration: 921.922 ms
     Changes:
----------
          ID: start mysql
    Function: service.running
        Name: mariadb
      Result: True
     Comment: The service mariadb is already running
     Started: 20:13:49.600782
    Duration: 39.01 ms
     Changes:
----------
          ID: service beacon example
    Function: beacon.present
        Name: service
      Result: True
     Comment: Adding service to beacons
              Beacon service saved
     Started: 20:13:49.642096
    Duration: 1412.349 ms
     Changes:
----------
          ID: disk usage beacon
    Function: beacon.present
        Name: diskusage
      Result: True
     Comment: Adding diskusage to beacons
              Beacon diskusage saved
     Started: 20:13:51.054599
    Duration: 41.851 ms
     Changes:
----------
          ID: install reactors
    Function: file.managed
        Name: /etc/salt/master.d/reactor.conf
      Result: True
     Comment: File /etc/salt/master.d/reactor.conf is in the correct state
     Started: 20:13:51.098898
    Duration: 147.079 ms
     Changes:

Summary for local
------------
Succeeded: 5
Failed:    0
------------
Total states run:     5
Total run time:   2.562 s


# salt-call beacons.list
local:
    beacons:
      diskusage:
      - enable: true
      - /: 50%
      - interval: 10
      service:
      - enable: true
      - services:
          mariadb:
            onchangeonly: true
      - interval: 5

# cat /etc/salt/minion.d/beacons.conf
beacons:
  beacons: {}

```
3.  No errors in the log...
```
# sleep 10; grep -c 'Unable to process beacon beacons' /var/log/salt/minion
0
```
4. Restart the minion and check for errors...
```
# systemctl restart salt-minion; sleep 10; grep -c 'Unable to process beacon beacons' /var/log/salt/minion
10
```

Both of the state files that are installing beacons have _save: True_ in them; here they are...

Here's the states that are running...

```
include: 
  - .mariadb
 
service beacon example:
  beacon.present:
    - name: service
    - enable: True
    - save: True
    - services:  
        mariadb:
          onchangeonly: True
    - interval: 5
    - require: 
      - start mysql

....
disk usage beacon:
  beacon.present:
    - name: diskusage
    - enable: True
    - save: True
    - '/': 50%
    - interval: 10
```

So your beacons are likely to trigger right up until you restart your minion.

**Comment by garethgreenaway** [comment]
Same issue as https://github.com/saltstack/salt/issues/58006 and fixed in #58854

**Comment by sagetherage** [comment]
Apologies, I thought this was a different issue and then realized my mistake, label and re-labeling.
